util/containerdutil/manifest.go

```diff
@@ -109,3 +116,43 @@ func ManifestDesc(ctx context.Context, provider content.Provider, image ocispec.
 	}
 	return m[0], nil
 }
+
+// Forked from github.com/containerd/containerd/images/image.go
```

Comment:
Is it worth exposing the original implementation rather than copying?  This logic covers the known cases of OCI indices, OCI manifests, Docker manifest lists, Docker schema 2 manifests, and Docker schema 1 manifests, but may have to change in the future if new types are introduced.

Reply:
Opened PR https://github.com/containerd/containerd/pull/6266